EventGenerator/src/DIOGenerator_tool.cc

```diff
@@ -29,43 +29,54 @@ namespace mu2e {
     explicit DIOGenerator(Parameters const& conf) :
       _pdgId(PDGCode::e_minus),
       _mass(GlobalConstantsHandle<ParticleDataTable>()->particle(_pdgId).ref().mass().value()),
-      _genId(GenId::DIOGenTool),
       _spectrum(BinnedSpectrum(conf().spectrum.get<fhicl::ParameterSet>()))
-    {
-
-    }
+    {}
 
+    std::vector<ParticleGeneratorTool::Kinematic> generate() override;
     void generate(std::unique_ptr<GenParticleCollection>& out, const IO::StoppedParticleF& stop) override;
 
-    void setEngine(art::RandomNumberGenerator::base_engine_t& eng) {
+    void finishInitialization(art::RandomNumberGenerator::base_engine_t& eng, const std::string&) override {
```

Comment:
Why is the material name unused?  DIO spectrum does depend on material, I guess that's a future development?  If so, please add a comment and a TODO flag